### PR TITLE
Decompile wall clock routines

### PR DESCRIPTION
**Pull request by camthesaxman**
I still can't figure out how to write the DMA code, and negating the sine at lines 468 and 501 doesn't produce identical code. However, the rest of it compiles correctly.

### CODE REVIEW

**Comment by YamaArashi** [comment]
Have you tried using the DmaFill32 macro?

**Comment by YamaArashi** [comment]
Is there any part that still doesn't match?

**Comment by camthesaxman** [comment]
Some of the operations and registers are reordered in LoadWallClockGraphics, particularly around the DmaFill part, and I can't seem to get the neg instruction to be in the right place in sub_810B05C and sub_810B0F4. I think everything else matches.

**Comment by YamaArashi** [comment]
This may work.

`
void LoadWallClockGraphics(void)
{
    u8 *addr;
    u32 size;

    SetVBlankCallback(0);
    REG_DISPCNT = 0;
    REG_BG3CNT = 0;
    REG_BG2CNT = 0;
    REG_BG1CNT = 0;
    REG_BG0CNT = 0;
    REG_BG3HOFS = 0;
    REG_BG3VOFS = 0;
    REG_BG2HOFS = 0;
    REG_BG2VOFS = 0;
    REG_BG1HOFS = 0;
    REG_BG1VOFS = 0;
    REG_BG0HOFS = 0;
    REG_BG0VOFS = 0;

    addr = VRAM;
    size = 0x18000;
    while (1)
    {
        DmaFill16(3, 0, addr, 0x1000);
        addr += 0x1000;
        size -= 0x1000;
        if (size <= 0x1000)
        {
            DmaFill16(3, 0, addr, size);
            break;
        }
    }

    {
        void *oam = (void *)OAM;
        u32 oamSize = OAM_SIZE;
        DmaFill32(3, 0, oam, oamSize);
    }

    {
        void *pltt = (void *)PLTT;
        u32 plttSize = PLTT_SIZE;
        DmaFill16(3, 0, pltt, plttSize);
    }
    
    LZ77UnCompVram(gMiscClock_Gfx, VRAM);
    if(gUnknown_0202E8CC)
        LoadPalette(gMiscClockFemale_Pal, 0, 32);
    else
        LoadPalette(gMiscClockMale_Pal, 0, 32);
    remove_some_task();
    ResetTasks();
    ResetSpriteData();
    ResetPaletteFade();
    FreeAllSpritePalettes();
    LoadCompressedObjectPic(gUnknown_083F7A90);
    LoadSpritePalettes(&gUnknown_083F7AA0);
    SetUpWindowConfig(&gWindowConfig_81E6C3C);
    InitMenuWindow(&gWindowConfig_81E6CE4);
}
`

**Comment by YamaArashi** [comment]
Try this. Also, add `#include "trig.h"` to the top. I just added trig.h.

```
void sub_810B05C(struct Sprite *sprite)
{
    u16 angle;
    s16 sin;
    s16 cos;
    u16 x;
    u16 y;
    
    angle = gTasks[sprite->data0].data[TD_MHAND_ANGLE];
    sin = Sin2(angle) / 16;
    cos = Cos2(angle) / 16;
    SetOamMatrix(0, cos, sin, -sin, cos);
    x = gClockHandCoords[angle][0];
    y = gClockHandCoords[angle][1];
    
    //Manual sign extension - why?
    if(x > 0x80)
        x |= 0xFF00;
    if(y > 0x80)
        y |= 0xFF00;
    
    sprite->pos2.x = x;
    sprite->pos2.y = y;
}
```

**Comment by camthesaxman** [comment]
Thanks a lot! How did you figure that out? The while(1) loop is less representative of the assembly, yet it compiles exactly. There's probably a macro we need to implement for those DmaFills in the curly braces. sub_810B05C, sub_810B0F4, sub_810B18C, and sub_810B230 appear to be unreferenced.
It now compiles exactly, but I need to rebase with the main branch.

**Comment by camthesaxman** [comment]
Okay, it's ready to be merged, unless there's anything you want me to change.

**Comment by YamaArashi** [comment]
```
ROM:0810A76C                 MOVS            R6, #0
ROM:0810A76E                 LDR             R1, =dword_40000D4
ROM:0810A770                 MOVS            R5, #0x1000
ROM:0810A774                 LDR             R7, =0x81000800
ROM:0810A776                 MOVS            R0, #0x81000000
ROM:0810A77A                 MOV             R12, R0
ROM:0810A77C
ROM:0810A77C loc_810A77C:                            @ CODE XREF: LoadWallClockGraphics+76j
ROM:0810A77C                 STRH            R6, [R2]
ROM:0810A77E                 MOV             R0, SP
ROM:0810A780                 STR             R0, [R1]
ROM:0810A782                 STR             R3, [R1,#(dword_40000D8 - 0x40000D4)]
ROM:0810A784                 STR             R7, [R1,#(dword_40000DC - 0x40000D4)]
ROM:0810A786                 LDR             R0, [R1,#(dword_40000DC - 0x40000D4)]
ROM:0810A788                 ADDS            R3, R3, R5
ROM:0810A78A                 SUBS            R4, R4, R5
ROM:0810A78C                 CMP             R4, R5
ROM:0810A78E                 BHI             loc_810A77C
ROM:0810A790                 STRH            R6, [R2]
ROM:0810A792                 MOV             R2, SP
```

I noticed that R6 was being reused as the "fill" value for the DMA. This looked like a CSE optimization, but they can't occur across loop boundaries. So, I figured that the DMA fill to mop up the leftover amount had to be inside the loop body.

File: src/wallclock.c [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+}
+
+//Determines the clock period (AM/PM) based on the hour
+static void UpdateClockPeriod(u8 taskId, u8 direction)
+{
+    u8 hours = gTasks[taskId].data[TD_HOURS];
+    
+    switch(direction)
+    {
+        case 1:
```
Comment (PikalaxALT) [inline src/wallclock.c]: For consistency, maybe use FORWARD and BACKWARD here as well?

Comment (camthesaxman) [inline src/wallclock.c]: Good catch!

**Pull request merged**